### PR TITLE
Fix/62692 community events attendees check in

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/62692

This PR adds an `event_ID` attribute on the attendees table when shown on an event context.
That's then used to conditionally reset the event attendees table cache for the event.

### CODE REVIEW

File: src/Tribe/Attendees_Table.php [hunk 1]
```diff
@@ -259,9 +259,14 @@ public function column_check_in( $item ) {
 
 			return sprintf( $button_template, $item['order_id_link_src'], __( 'View order', 'event-tickets' ) );
 		}
-
-		$checkin   = sprintf( '<a href="#" data-attendee-id="%d" data-provider="%s" class="button-secondary tickets_checkin">%s</a>', esc_attr( $item['attendee_id'] ), esc_attr( $item['provider'] ), esc_html__( 'Check in', 'event-tickets' ) );
-		$uncheckin = sprintf( '<span class="delete"><a href="#" data-attendee-id="%d" data-provider="%s" class="tickets_uncheckin">%s</a></span>', esc_attr( $item['attendee_id'] ), esc_attr( $item['provider'] ), esc_html__( 'Undo Check in', 'event-tickets' ) );
+		if (empty($this->event)) {
```
Comment (barryhughes) [inline src/Tribe/Attendees_Table.php]: Spacing!

`if ( empty( $this->event ) ) {`

File: src/Tribe/Tickets.php [hunk 2]
```diff
@@ -1389,5 +1393,15 @@ public static function get_login_url() {
 			 */
 			return apply_filters( 'tribe_tickets_ticket_login_url', $login_url );
 		}
+
+		/**
+		 * @param $return
```
Comment (barryhughes) [inline src/Tribe/Tickets.php]: What's the purpose of `$return` - could we add a short explanation? Not for instance quite understanding why we wouldn't do something like this at the callsite:

```
if ( $this->uncheckin( $order_id ) ) { 
    $this->maybe_update_attendees_cache(); 
}
```

Comment (lucatume) [inline src/Tribe/Tickets.php]: if the check-in or uncheck-in failed then there is no use in re-updating the cache, `return` will be `true` or `false` to reflect that

Comment (barryhughes) [inline src/Tribe/Tickets.php]: Hmm, ok. Maybe it's just me but `$return` doesn't seem like the best name in that case (it possibly makes more sense at the call site, ie in the `has_permission()` method) but right here it doesn't really reflect what it does.

Comment (lucatume) [inline src/Tribe/Tickets.php]: It was the pre-existing var name, but I agree that's not very explicit... something like `did_check_in` e `did_uncheck_in`?

Comment (lucatume) [inline src/Tribe/Tickets.php]: How about `did_check_in` and `did_uncheck_in`?

File: src/Tribe/Tickets.php [hunk 3]
```diff
@@ -654,7 +658,7 @@ public static function get_event_attendees( $event_id ) {
 			$attendees = array();
 			if ( ! is_admin() ) {
 				$post_transient = Tribe__Post_Transient::instance();
-
+
```
Comment (borkweb) [inline src/Tribe/Tickets.php]: A bunch of stray whitespace was added here.

**Comment by borkweb** [comment]
Get rid of the stray whitespace and this is good to go!

**Comment by lucatume** [comment]
Thanks matt!

**Pull request merged**